services/slam/fake/slam_test.go

```diff
@@ -0,0 +1,173 @@
+package fake
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+func TestFakeSLAMPosition(t *testing.T) {
+	slamSvc := &FakeSLAM{Name: "test"}
+	pInFrame, err := slamSvc.Position(context.Background(), slamSvc.Name, map[string]interface{}{})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pInFrame.Parent(), test.ShouldEqual, slamSvc.Name)
+
+	expectedPoint := r3.Vector{X: 1.8607751801785188, Y: 34.26593374183797, Z: 0}
+	test.That(t, pInFrame.Pose().Point(), test.ShouldResemble, expectedPoint)
+
+	expectedOri := spatialmath.NewR4AA()
+	expectedOri.RZ = -1
+	expectedOri.Theta = 3.0542483867902197
+	test.That(t, pInFrame.Pose().Orientation().AxisAngles(), test.ShouldResemble, expectedOri)
+
+	pInFrame2, err := slamSvc.Position(context.Background(), slamSvc.Name, map[string]interface{}{})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pInFrame2, test.ShouldResemble, pInFrame2)
+}
+
+func TestFakeSLAMGetInternalState(t *testing.T) {
+	slamSvc := &FakeSLAM{Name: "test"}
+	data, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, len(data), test.ShouldBeGreaterThan, 0)
+	data2, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, data, test.ShouldResemble, data2)
+}
+
+func TestFakeSLAMStateful(t *testing.T) {
+	t.Run("Test getting a PCD map advances the test data", func(t *testing.T) {
+		slamSvc := &FakeSLAM{Name: "test"}
+		extra := map[string]interface{}{}
+		verifyGetMapStateful(t, rdkutils.MimeTypePCD, slamSvc, extra)
+	})
+
+	t.Run("Test getting a JPEG map advances the test data", func(t *testing.T) {
+		slamSvc := &FakeSLAM{Name: "test"}
+		extra := map[string]interface{}{}
+		verifyGetMapStateful(t, rdkutils.MimeTypeJPEG, slamSvc, extra)
```

Comment:
As a note, this test takes ~30 seconds on my laptop due to the overhead of encoding & decoding jpegs.

Reply:
Just curious why the cartographer maps are saved as pngs instead of jpegs. If we saved as jpegs, decoding would be cheaper, and we might even be able to return a lazy jpeg from GetMap and avoid doing any decoding/encoding.

Reply:
These are what Jeremy collected. I can convert them to jpegs & see if that speeds up the test suite.

Reply:
@JohnN193 has recommended to me offline to not convert them to JPEGS as doing so would hide the fact that currently, the vision/image encode/decode code is the source of these inefficiencies, which are also present in the live version of GetMap.

Reply:
I don't think it's important, since we're removing GetMap soon.

Reply:
to be clear, cartographer does **not** return PNGs, and I am not sure why these were saved as PNGs initially. I did not believe it was necessary to convert the PNGs Jeremy recorded to JPEGs though for this implementation

Reply:
I'm going to leave this data set as is given that this endpoint is going away anyway & I'd really like to get this PR merged ASAP so I don't block frontend 2d map work on this PR anymore.